setup.bzl

```diff
@@ -0,0 +1,149 @@
+
+def _include_if_not_defined(repo_rule, name, **kwargs):
+    if name not in native.existing_rules():
+        repo_rule(name = name, **kwargs)
+
+JINJA2_BUILD_FILE = """
+py_library(
+    name = "jinja2",
+    srcs = glob(["jinja2/*.py"]),
+    srcs_version = "PY2AND3",
+    deps = [
+        "@markupsafe_archive//:markupsafe",
+    ],
+    visibility = ["//visibility:public"],
+)
+"""
+
+MARKUPSAFE_BUILD_FILE = """
+py_library(
+    name = "markupsafe",
+    srcs = glob(["markupsafe/*.py"]),
+    srcs_version = "PY2AND3",
+    visibility = ["//visibility:public"],
+)
+"""
+
+MISTUNE_BUILD_FILE = """
+py_library(
+    name = "mistune",
+    srcs = ["mistune.py"],
+    srcs_version = "PY2AND3",
+    visibility = ["//visibility:public"],
+)
+"""
+
+SIX_BUILD_FILE = """
+py_library(
+    name = "six",
+    srcs = ["six.py"],
+    srcs_version = "PY2AND3",
+    visibility = ["//visibility:public"],
+)
+"""
+
+def skydoc_repositories():
+    """Adds the external repositories used by the skylark rules."""
+    _include_if_not_defined(
+        git_repository,
+        name = "bazel_skylib",
+        remote = "https://github.com/bazelbuild/bazel-skylib.git",
+        tag = "0.6.0",
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel_rules_sass",
+        remote = "https://github.com/bazelbuild/rules_sass.git",
+        tag = "1.15.1",
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel_rules_sass",
+        remote = "https://github.com/bazelbuild/rules_sass.git",
+        tag = "1.15.1",
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel",
+        remote = "https://github.com/bazelbuild/bazel.git",
+        # TODO: Update to a newer tagged version when available.
+        commit = "e7ebb7e68d35ae090d91fe6b4c92c1c831421faa",  # 2018-11-26
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "com_google_protobuf",
+        remote = "https://github.com/protocolbuffers/protobuf.git",
+        # Latest tagged version at time of writing is v3.6.1, which doesn't
+        # include fixes for --incompatible_package_name_is_a_function,
+        # --incompatible_new_actions_api, and possibly others.
+        # TODO: Update to a newer tagged version when available.
+        commit = "7b28271a61a3da0a37f6fda399b0c4c86464e5b3",  # 2018-11-16
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel_rules_sass",
+        remote = "https://github.com/bazelbuild/rules_sass.git",
+        tag = "1.15.1",
+    )
+
+    # Protobuf expects an //external:python_headers label which would contain the
+    # Python headers if fast Python protos is enabled. Since we are not using fast
+    # Python protos, bind python_headers to a dummy target.
+    native.bind(
+        name = "python_headers",
+        actual = "//:dummy",
+    )
+
+    http_archive(
```

Comment:
I think you should add these repositories conditionally as well. While you're at it, may as well let your users overwrite them

Reply:
Done.